internal/services/machinelearning/machine_learning_registry_resource.go

```diff
@@ -0,0 +1,555 @@
+
+								"system_created_storage_account_hns_enabled": {
+									Type:     pluginsdk.TypeBool,
+									Optional: true,
+									Default:  false,
+								},
+
+								"system_created_storage_account_arm_resource_id": {
+									Type:     pluginsdk.TypeString,
+									Computed: true,
+								},
+
+								"system_created_storage_account_name": {
+									Type:     pluginsdk.TypeString,
+									Computed: true,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+
+		"tags": commonschema.Tags(),
+	}
+}
+
+func (m MachineLearningRegistryResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"discovery_url": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+
+		"intellectual_property_publisher": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+
+		"ml_flow_registry_uri": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+
+		// API will return a generated mrg even provided in request
+		"managed_resource_group": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+	}
+}
+
+func (m MachineLearningRegistryResource) ModelObject() interface{} {
+	return &MachineLearningRegistryModel{}
+}
+
+func (m MachineLearningRegistryResource) ResourceType() string {
+	return "azurerm_machine_learning_registry"
+}
+
+func (m MachineLearningRegistryResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, meta sdk.ResourceMetaData) error {
+			client := meta.Client.MachineLearning.RegistryManagementClient
+
+			var model MachineLearningRegistryModel
+			if err := meta.Decode(&model); err != nil {
+				return err
+			}
+
+			subscriptionID := meta.Client.Account.SubscriptionId
+			id := registrymanagement.NewRegistryID(subscriptionID, model.ResourceGroupName, model.Name)
+			existing, err := client.RegistriesGet(ctx, id)
+			if !response.WasNotFound(existing.HttpResponse) {
+				if err != nil {
+					return fmt.Errorf("retrieving %s: %v", id, err)
+				}
+				return meta.ResourceRequiresImport(m.ResourceType(), id)
+			}
+
+			req := registrymanagement.RegistryTrackedResource{
+				Location: model.Location,
+				Name:     &model.Name,
+				Tags:     &model.Tags,
+			}
+
+			if req.Identity, err = identity.ExpandLegacySystemAndUserAssignedMapFromModel(model.Identity); err != nil {
+				return fmt.Errorf("expand identity: %+v", err)
+			}
+
+			req.Properties = registrymanagement.Registry{
+				PublicNetworkAccess:  publicNetworkAccessFromBool(model.PublicNetworkAccess),
+				ManagedResourceGroup: m.expandArmResourceID(model.ManagedResourceGroup),
+				RegionDetails:        m.expandRegionDetails(model.RegionDetails),
+			}
+
+			if err = client.RegistriesCreateOrUpdateThenPoll(ctx, id, req); err != nil {
+				return fmt.Errorf("creating %s: %v", id, err)
+			}
+
+			meta.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (m MachineLearningRegistryResource) Update() sdk.ResourceFunc {
```

Comment:
The additional regions, now renamed to `replica_locations` in the schema, should be updatable, or in the very least, removable. Can you please investigate?